src/PKSim.Infrastructure/Serialization/Xml/Serializers/UsedBuildingBlockXmlSerializer.cs

```diff
@@ -33,6 +33,9 @@ protected override void TypedDeserialize(UsedBuildingBlock usedBuildingBlock, XE
          var buildingBlockNode = usedBuildingBlockNode.Elements().First();
          var serializer = SerializerRepository.SerializerFor(buildingBlockNode);
 
+         if (serializer == null)
```

Comment:
hum... that's probably some serious shit going wrong here. Or why would you accept null? If this is a valid use casse, maybe comment why?

Reply:
This is when you have the element and you are trying to find the serializer. If the element is not a known type, then it should not throw an exception. That change is made also in OSPSuite.Serializer.

Reply:
Or maybe we should leave this one as throwing an exception?

Reply:
because heere we would have a UsedBuildingBlock WITHOUT a building block. That feels like it should NEVER happen

Reply:
Yeah, that's right. We usually don't do this Elements.First() and then try to get the serializer. This is unique.

I guess this means that the order of the XML is important. Maybe an issue should be entered for that.

Reply:
Why wouldn't there be 

```
Map(x => x.BuildingBlock);
```

Reply:
ah, it's because the actual building block type is unknown I guess.

Reply:
yeah exactly. That's definitely a bit weird
The other options would be to make sure the tool does not crash if the ref to the BB is not found

Reply:
We can just throw the serializer not found exception as it would have done before with old ospsuite.serializer.

Reply:
I changed it to throw the Serializer not found exception. That's what would have happened in previous OSPSuite.Serializer if a serializer can't be found from the element.